packages/react/src/components/Accordion/Accordion.tsx

```diff
@@ -146,33 +197,73 @@ class Accordion extends AutoControlledComponent<WithAsProp<AccordionProps>, any>
       _.invoke(predefinedProps, 'onClick', e, titleProps)
       _.invoke(this.props, 'onTitleClick', e, titleProps)
     },
+    onFocus: (e: React.SyntheticEvent, titleProps: AccordionTitleProps) => {
+      _.invoke(predefinedProps, 'onFocus', e, titleProps)
+      this.setState({ focusedIndex: titleProps.index })
+    },
   })
 
-  isIndexActive = (index): boolean => {
+  private isIndexActive = (index: number): boolean => {
     const { exclusive } = this.props
     const { activeIndex } = this.state
 
     return exclusive ? activeIndex === index : _.includes(activeIndex, index)
   }
 
+  private isIndexClosingPrevented = (index: number): boolean => {
+    const { activeIndex } = this.state
+    const { alwaysActive, exclusive } = this.props
+
+    if (alwaysActive) {
+      if (exclusive) {
+        return activeIndex === index
+      }
+      return activeIndex.length === 1 && activeIndex[0] === index
+    }
+
+    return false
+  }
+
   renderPanels = () => {
     const children: any[] = []
     const { panels, renderPanelContent, renderPanelTitle } = this.props
+    const { focusedIndex } = this.state
+
+    this.itemRefs = []
+    this.focusHandler.syncFocusedIndex(focusedIndex)
 
     _.each(panels, (panel, index) => {
       const { content, title } = panel
       const active = this.isIndexActive(index)
+      const cannotBeClosed = this.isIndexClosingPrevented(index)
+      const buttonRef = React.createRef<HTMLElement>()
+      const titleId = title['id'] || _.uniqueId('accordion-title-')
+      const contentId = content['id'] || _.uniqueId('accordion-content-')
+      this.itemRefs[index] = buttonRef
```

Comment:
do you need a `buttonRef` prop in `AccordionTitle` component?
I mean, to do it in the same way how it's done in List.

```jsx
const ref = React.createRef()
this.itemRefs[index] = ref
...
const accordionTitleProps = {
   ...
   ref,
   ...
}

AccordionTitle.create(title, {
    defaultProps: accordionTitleProps,
    ...
})
```

Reply:
isn't that going to give me the `ref` to the `<ElementType>` ? I need the one from `<Button>` which is the child of `<ElementType>`

I am applying navigation on the buttons, not on their parents.